lib/pages/child/plan_in_progress_page.dart

```diff
@@ -18,181 +31,232 @@ class _ChildPlanInProgressPageState extends State<ChildPlanInProgressPage> {
-				ItemCard(
-					title: "Opróżnij plecak",
-					chips:
-					<Widget>[
-						AttributeChip.withIcon(
-							icon: Icons.timer,
-							color: AppColors.childBackgroundColor,
-							content: "Czas: 2:10"
-						)
-					],
-					actionButton: ItemCardActionButton(
-							color: AppColors.childBackgroundColor, icon: Icons.check, onTapped: () => log("Tapped finished activity")
-					),
-				),
-				ItemCard(
-					title: "Przygotuj książki i zeszyty na kolejny dzień według bardzo długiego planu zajęć",
-					chips:<Widget>[
-						AttributeChip.withIcon(
-							icon: Icons.access_time,
-							color: AppColors.childActionColor,
-							content: "6 minut"
-						)
-					],
-					actionButton:	ItemCardActionButton(color: Colors.teal, icon: Icons.play_arrow, onTapped: () => log("tapped start task")),
-				),
-				ItemCard(
-					title: "Spakuj potrzebne rzeczy",
-					chips:
-					<Widget>[
-						AttributeChip.withIcon(
-							icon: Icons.access_time,
-							color: AppColors.childActionColor,
-							content: "5 minut"
-						)
-					],
-					actionButton: ItemCardActionButton(color: Colors.grey, icon: Icons.keyboard_arrow_up),
-				),
-				ItemCard(
-					title: "Spakuj potrzebne rzeczy",
-					chips:
-					<Widget>[
-						AttributeChip.withIcon(
-							icon: Icons.access_time,
-							color: AppColors.childActionColor,
-							content: "5 minut"
+				for(var task in tasks)
+					if(task.taskUiType == TaskUIType.completed)
+						getCompletedTaskCard(task)
+					else if(task.taskUiType == TaskUIType.available)
+						ItemCard(
+							title: task.name,
+							subtitle: task.description,
+							chips:<Widget>[
+								if (task.timer > 0) getTimeChip(task),
+								if (task.points.quantity != 0) getCurrencyChip(task)
+							],
+							actionButton:	ItemCardActionButton(color: Colors.teal, icon: Icons.play_arrow, onTapped: () => log("tapped start task")),
 						)
-					],
-					actionButton: ItemCardActionButton(color: Colors.grey, icon: Icons.keyboard_arrow_up),
-				),
-				ItemCard(
-					title: "Spakuj potrzebne rzeczy",
-					chips:
-					<Widget>[
-						AttributeChip.withIcon(
-							icon: Icons.access_time,
-							color: AppColors.childActionColor,
-							content: "5 minut"
+					else if(task.taskUiType == TaskUIType.stopped)
+							ItemCard(
+								title: task.name,
+								subtitle: task.description,
+								chips:<Widget>[
+									getDurationChip(task),
+									getBreaksChip(task),
+									if (task.timer > 0) getTimeChip(task),
+									if (task.points.quantity != 0) getCurrencyChip(task)
+								],
+								actionButton:	ItemCardActionButton(color: Colors.teal, icon: Icons.play_arrow, onTapped: () => log("tapped start task")),
+							)
+					else if(task.taskUiType.inProgress)
+						BlocProvider<TimerCubit>(
+							create: (_) => TimerCubit(() => task.taskUiType == TaskUIType.currentlyPerformed ? sumDurations(task.duration).inSeconds : sumDurations(task.breaks).inSeconds)..startTimer(),
```

Comment:
I see you created the elapsedTimePassed field but you are not using it :)

Reply:
DAMN